### PR TITLE
Remove support for EKS 1.22

### PR DESCRIPTION
**Pull request by watany-dev**
### Description

Description
Remove support for EKS 1.22.

Resolves https://github.com/weaveworks/eksctl/issues/6202.

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [X] Manually tested
- [X] Made sure the title of the PR is a good description that can go into the release notes
- [X] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/outposts/cluster_extender.go [hunk 1]
```diff
@@ -31,8 +31,9 @@ type stackUpdater interface {
 	AppendNewClusterStackResource(ctx context.Context, extendForOutposts, plan bool) (bool, error)
 }
 
-//counterfeiter:generate -o fakes . ClusterToExtend
 // ClusterToExtend represents a cluster that needs to be extended.
+//
```
Comment (Himangini) [inline pkg/outposts/cluster_extender.go:35]: Can you pls remove changes like these from the PR?

File: pkg/apis/eksctl.io/v1alpha5/generate.go [hunk 2]
```diff
@@ -7,6 +7,7 @@ import (
 
 //go:generate go run ../../../../cmd/schema assets/schema.json
 
-//SchemaJSON contains the schema
+// SchemaJSON contains the schema
```
Comment (Himangini) [inline pkg/apis/eksctl.io/v1alpha5/generate.go:10]: pls, remove the changes to the comments to make the review diff small.

File: pkg/actions/cluster/upgrade_test.go [hunk 3]
```diff
@@ -55,9 +55,9 @@ var _ = Describe("upgrade cluster", func() {
 		}),
 
 		Entry("upgrades to the next version when specified", upgradeCase{
-			givenVersion:           "1.22",
-			eksVersion:             "1.21",
-			expectedUpgradeVersion: "1.22",
+			givenVersion:           "1.23",
+			eksVersion:             "W",
```
Comment (Himangini) [inline pkg/actions/cluster/upgrade_test.go:59]: "W"?

**Review by Himangini — CHANGES_REQUESTED** [review]
pls rebase to remove conflicts, also we have another PR to bump deps so these changes can be rebased again once the other PR gets merged. See comments below.

File: pkg/outposts/cluster_extender.go (hunk 1 above)
Comment (watany-dev) [reply pkg/outposts/cluster_extender.go]: thx! fixed.

File: pkg/apis/eksctl.io/v1alpha5/generate.go (hunk 2 above)
Comment (watany-dev) [reply pkg/apis/eksctl.io/v1alpha5/generate.go]: thx! fixed.

File: pkg/actions/cluster/upgrade_test.go (hunk 3 above)
Comment (watany-dev) [reply pkg/actions/cluster/upgrade_test.go]: Typing error. Fixed.

File: integration/matchers/kubernetes.go [hunk 4]
```diff
@@ -9,8 +9,8 @@ import (
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 )
 
-//BeNotFoundError succeeds if actual is a non-nil error
-//which represents a missing kubernetes resource
+// BeNotFoundError succeeds if actual is a non-nil error
+// which represents a missing kubernetes resource
```
[the remark this answers is not in the document]
Comment (watany-dev) [reply integration/matchers/kubernetes.go:13]: fixed.

File: pkg/cfn/builder/vpc.go [hunk 5]
```diff
@@ -45,7 +45,7 @@ const (
 	defaultDesiredMaskSize = defaultSubnetMask - defaultPrefix
 )
 
-//VPCResourceSet interface for creating cloudformation resource sets for generating VPC resources
+// VPCResourceSet interface for creating cloudformation resource sets for generating VPC resources
 type VPCResourceSet interface {
```
[the remark this answers is not in the document]
Comment (watany-dev) [reply pkg/cfn/builder/vpc.go:49]: fixed

File: pkg/credentials/filecache.go [hunk 6]
```diff
@@ -22,8 +22,9 @@ const (
 	EksctlCacheFilenameEnvName = "EKSCTL_CREDENTIAL_CACHE_FILENAME"
 )
 
-//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
 // Clock implements Now to return the current time.
+//
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
 //counterfeiter:generate -o fakes/fake_clock.go . Clock
```
[the remark this answers is not in the document]
Comment (watany-dev) [reply pkg/credentials/filecache.go:28]: fixed.

File: pkg/karpenter/karpenter.go [hunk 7]
```diff
@@ -39,6 +39,7 @@ type Options struct {
 }
 
 // ChartInstaller defines a functionality to install Karpenter.
+//
 //go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
```
[the remark this answers is not in the document]
Comment (watany-dev) [reply pkg/karpenter/karpenter.go:43]: fixed,

File: pkg/kubernetes/client.go [hunk 8]
```diff
@@ -497,7 +497,7 @@ func (r *RawResource) Get() (runtime.Object, bool, error) {
 	return obj, true, nil
 }
 
-//Credit https://github.com/helm/helm/issues/6910#issuecomment-601277026
+// Credit https://github.com/helm/helm/issues/6910#issuecomment-601277026
 type SimpleRESTClientGetter struct {
```
[the remark this answers is not in the document]
Comment (watany-dev) [reply pkg/kubernetes/client.go:501]: fixed.

File: pkg/nodebootstrap/assets/assets.go [hunk 9]
```diff
@@ -5,30 +5,37 @@ import (
 	_ "embed"
 )
 
-//BootstrapAl2Sh holds the bootstrap.al2.sh contents
+// BootstrapAl2Sh holds the bootstrap.al2.sh contents
+//
 //go:embed scripts/bootstrap.al2.sh
```
[the remark this answers is not in the document]
Comment (watany-dev) [reply pkg/nodebootstrap/assets/assets.go:10]: fixed

**Comment by watany-dev** [comment]
All unnecessary fixes that occurred in go fmt have been restored.

**Review by Himangini — COMMENTED** [review]
The unit tests are failing at 
```
[FAIL] default addons - coredns UpdateCoreDNS [It] updates coredns to the correct version
  /home/runner/work/eksctl/eksctl/pkg/addons/default/coredns_test.go:66
  ```

**Review by Himangini — COMMENTED** [review]
Thanks for fixing all the issues, PR LGTM. Going to run integration tests before merging.

**Review by Himangini — APPROVED** [review]
LGTM 👍🏻 Integration tests passed.
Thanks for your contribution ✨

**Pull request merged**